internal/sbi/consumer/nrf_service.go

```diff
@@ -0,0 +1,302 @@
+	pcf_context "github.com/free5gc/pcf/internal/context"
+	"github.com/free5gc/pcf/internal/logger"
+	"github.com/free5gc/pcf/internal/util"
+)
+
+type nnrfService struct {
+	consumer *Consumer
+
+	nfMngmntMu sync.RWMutex
+	nfDiscMu   sync.RWMutex
+
+	nfMngmntClients map[string]*Nnrf_NFManagement.APIClient
+	nfDiscClients   map[string]*Nnrf_NFDiscovery.APIClient
+}
+
+func (s *nnrfService) getNFManagementClient(uri string) *Nnrf_NFManagement.APIClient {
+	if uri == "" {
+		return nil
+	}
+	s.nfMngmntMu.RLock()
+	client, ok := s.nfMngmntClients[uri]
+	if ok {
+		defer s.nfMngmntMu.RUnlock()
+		return client
+	}
+
+	configuration := Nnrf_NFManagement.NewConfiguration()
+	configuration.SetBasePath(uri)
+	client = Nnrf_NFManagement.NewAPIClient(configuration)
+
+	s.nfMngmntMu.RUnlock()
+	s.nfMngmntMu.Lock()
+	defer s.nfMngmntMu.Unlock()
+	s.nfMngmntClients[uri] = client
+	return client
+}
+
+func (s *nnrfService) getNFDiscClient(uri string) *Nnrf_NFDiscovery.APIClient {
+	if uri == "" {
+		return nil
+	}
+	s.nfDiscMu.RLock()
+	client, ok := s.nfDiscClients[uri]
+	if ok {
+		defer s.nfDiscMu.RUnlock()
+		return client
+	}
+
+	configuration := Nnrf_NFDiscovery.NewConfiguration()
+	configuration.SetBasePath(uri)
+	client = Nnrf_NFDiscovery.NewAPIClient(configuration)
+
+	s.nfDiscMu.RUnlock()
+	s.nfDiscMu.Lock()
+	defer s.nfDiscMu.Unlock()
+	s.nfDiscClients[uri] = client
+	return client
+}
+
+func (s *nnrfService) SendSearchNFInstances(
+	nrfUri string, targetNfType, requestNfType models.NfType, param Nnrf_NFDiscovery.SearchNFInstancesParamOpts) (
+	*models.SearchResult, error,
+) {
+	// Set client and set url
+	client := s.getNFDiscClient(nrfUri)
+
+	ctx, _, err := pcf_context.GetSelf().GetTokenCtx(models.ServiceName_NNRF_DISC, models.NfType_NRF)
+	if err != nil {
+		return nil, err
+	}
+
+	result, res, err := client.NFInstancesStoreApi.SearchNFInstances(ctx, targetNfType, requestNfType, &param)
+	if err != nil {
+		logger.ConsumerLog.Errorf("SearchNFInstances failed: %+v", err)
+	}
+	defer func() {
+		if resCloseErr := res.Body.Close(); resCloseErr != nil {
+			logger.ConsumerLog.Errorf("NFInstancesStoreApi response body cannot close: %+v", resCloseErr)
+		}
+	}()
+	if res != nil && res.StatusCode == http.StatusTemporaryRedirect {
+		return nil, fmt.Errorf("Temporary Redirect For Non NRF Consumer")
+	}
+
+	return &result, nil
+}
+
+func (s *nnrfService) SendNFInstancesUDR(nrfUri, id string) string {
+	targetNfType := models.NfType_UDR
+	requestNfType := models.NfType_PCF
+	localVarOptionals := Nnrf_NFDiscovery.SearchNFInstancesParamOpts{
+		// 	DataSet: optional.NewInterface(models.DataSetId_SUBSCRIPTION),
+	}
+	// switch types {
+	// case NFDiscoveryToUDRParamSupi:
+	// 	localVarOptionals.Supi = optional.NewString(id)
+	// case NFDiscoveryToUDRParamExtGroupId:
+	// 	localVarOptionals.ExternalGroupIdentity = optional.NewString(id)
+	// case NFDiscoveryToUDRParamGpsi:
+	// 	localVarOptionals.Gpsi = optional.NewString(id)
+	// }
```

Comment:
hmm...

Reply:
@yccodr  Predecessors left these. Am I allowed to remove it?

Reply:
Make sure why the origin code is commented, 
If the code is useful, fix it; if not, remove it.